pkg/apiserver/apiserver.go

```diff
@@ -76,25 +83,36 @@ var (
 
 func init() {
 	prometheus.MustRegister(requestCounter)
+	prometheus.MustRegister(realTimeRequestCounter)
 	prometheus.MustRegister(requestLatencies)
 	prometheus.MustRegister(requestLatenciesSummary)
 }
 
-// monitor is a helper function for each HTTP request handler to use for
-// instrumenting basic request counter and latency metrics.
-func monitor(verb, resource *string, client string, httpCode *int, reqStart time.Time) {
+// postRequestMonitor is a helper function for each HTTP request handler to use for
+// instrumenting basic request counter and latency metrics. It's meant to be called before
+// the filter chain which leads to the actual request.
+func postRequestMonitor(verb, resource *string, client string, httpCode *int, reqStart time.Time) {
 	requestCounter.WithLabelValues(*verb, *resource, client, strconv.Itoa(*httpCode)).Inc()
 	requestLatencies.WithLabelValues(*verb, *resource, client).Observe(float64((time.Since(reqStart)) / time.Microsecond))
 	requestLatenciesSummary.WithLabelValues(*verb, *resource).Observe(float64((time.Since(reqStart)) / time.Microsecond))
+	realTimeRequestCounter.WithLabelValues(*verb, *resource, client).Dec()
+}
+
+// preRequestMonitor is a helper function for each HTTP request handler to use for
+// instrumenting basic request counter and latency metrics. It's meant to be called after
+// the filter chain, and hence after the request is finished.
+func preRequestMonitor(verb, resource *string, client string) {
```

Comment:
optional nit: a nicer interface is if you have this return an interface or function, so that people can do the start and stop all in one line, like ```defer preMonitor(params).stop()```